File: strategy/python/item.py
class Item:
    """
    Sepette bulunan ürünleri temsil eden sınıf.
    Ürün kodu ve fiyat bilgisi temel olarak yeterlidir.
    """

    def __init__(self, up_code, price):
        self._up_code = up_code
        self._price = price

    def get_price(self):
        return self._price

File: strategy/typescript/item.ts
// Sepette bulunan ürünleri temsil eden sınıf.
// Ürün kodu ve fiyat bilgileri temel olarak yeterlidir.
export class Item {
  constructor(
    private readonly upcCode: string,
    private readonly price: number,
  ) {}

  public getPrice(): number {
    return this.price;
  }
}
